proposals/kapp-controller/004-health-status-reporting/README.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Enhancing Resource Health Monitoring within App CR"
+authors: [ "Varsha Prasad Narsing <vnarsing@redhat.com>" ]
+status: "in review"
+approvers: [ "Carvel Maintainers" ]
+---
+
+# Enhancing Resource Health Monitoring within App CR
+
+## Problem Statement
+
+After applying the package contents fetched from the source to the cluster through the App CR, it becomes challenging to ascertain the real-time health status of individual resources within the cluster. This lack of visibility poses a significant challenge for SRE/ops teams tasked with overseeing hundreds or even thousands of clusters. Consequently, having this information integrated into the Kapp controller can establish a centralized and reliable source of truth for monitoring purposes.
+
+## Terminology / Concepts
+
+## Proposal
+
+We intend to extend the existing App API by adding a new status condition to expose the system's health. To do so, the following needs to be implemented:
+
+1. The controller reconciling the App CR needs to dynamically set up watches for the resources being deployed by the package. 
```

Comment:
This might not be ideal, but you can find all the resources by taking the involved GroupKinds from the ConfigMap associated with the kapp app, then listing/watching/informing using the appropriate app label selector.

Reply:
Another possible approach would be to have `kapp` write some information about specific resources to a file upon reconciliation and have this information copied over to the status.
This is how we have information about used group versions and namespaces to the App status, using output from the `--app-metadata-file-output` flag.

This however means this information will be reported whenever the App syncs.